src/main/java/com/rcv/ElectionConfig.java

```diff
@@ -59,23 +62,119 @@ boolean validate() {
     boolean isValid = true;
 
     Logger.executionLog(Level.INFO, "Validating config...");
-    // TODO: reorder all checks so they go in same order as fields in config GUI
-    // TODO: need to add checks that all required String fields !.equals("")
 
-    if (getNumDeclaredCandidates() == 0) {
+    if (getContestName() == null || getContestName().isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Contest name is required.");
+    }
+
+    if (getOutputDirectory() == null || getOutputDirectory().isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Output directory is required.");
+    }
+
+    if (rawConfig.cvrFileSources == null || rawConfig.cvrFileSources.isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Config doesn't contain any CVR files.");
+    } else {
+      HashSet<String> cvrFilePathSet = new HashSet<>();
+      for (CVRSource source : rawConfig.cvrFileSources) {
+        if (source.getFilePath() == null || source.getFilePath().isEmpty()) {
+          isValid = false;
+          Logger.executionLog(Level.SEVERE, "filePath is required for each CVR file.");
+        } else if (cvrFilePathSet.contains(source.getFilePath())) {
+          isValid = false;
+          Logger.executionLog(
+              Level.SEVERE, "Duplicate CVR filePaths are not allowed: %s", source.getFilePath());
```

Comment:
I agree with this, but FYI, it breaks our scale test.

Reply:
Also agree.  Good move.